tensorflow/compiler/xla/tools/interactive_graphviz.cc

```diff
@@ -475,16 +470,43 @@ void DoPlotCommand(const Options& opts, const HloModule& module,
   }
 
   uint64 graph_width = kDefaultWidth;
-  if (tokens.size() == 2) {
+  std::set<const HloInstruction*> boundary;
+  if (tokens.size() >= 2) {
     if (comp) {
       std::cerr << "Can only use graph-size parameter with instructions, but "
                 << node_name << " is a computation." << std::endl;
       return;
     }
+
+    int bound_index = tokens.size();
     if (!absl::SimpleAtoi(tokens[1], &graph_width)) {
-      std::cerr << "Can't parse '" << tokens[1] << "' as an integer."
-                << std::endl;
-      return;
+      if (tokens[1] != "/") {
+        std::cerr << "Can't parse '" << tokens[1] << "' as an integer."
+                  << std::endl;
+        return;
+      }
+      graph_width = kDefaultWidth;
+      bound_index = 2;
+    } else {
+      if (tokens.size() > 2) {
+        if (tokens[2] != "/") {
+          std::cerr << "Expect a /, but get a '" << tokens[1] << "'."
+                    << std::endl;
+          return;
+        }
+        bound_index = 3;
+      }
+    }
+    while (bound_index < tokens.size()) {
+      string bnode_name = tokens[bound_index];
+      const HloInstruction* binstr = FindInstruction(module, bnode_name);
+      if (!binstr) {
+        std::cerr << "Couldn't find HloInstruction named " << node_name << "."
+                  << std::endl;
+        return;
+      }
+      boundary.insert(binstr);
+      bound_index++;
```

Comment:
Hm, this logic is a bit confusing to me.  How about implementing something like the `shift` operator in bash?

```
void DoPlotCommand(const Options& opts, const HloModule& module, absl::Span<const string> tokens) {
  string node_name = tokens[0];
  tokens.remove_prefix(1);

  // Next token can either be a number (graph width) or "/".
  if (!tokens.empty() && absl::SimpleAtoi(tokens[0], &graph_width)) {
    tokens.remove_prefix(1);
  }

  // Next token (if present) must be "/", introducing the set of boundary
  // instructions.
  if (!tokens.empty()) {
    if (tokens[0] != "/") { std::cerr << "..."; return; }
    tokens.remove_prefix(1);
  }

  // All remaining tokens are boundary instructions.
  for (const string& boundary_instr_name : tokens) {
    ...
  }
}
```